pkg/backup/resource_backupper.go

```diff
@@ -237,43 +240,89 @@ func (rb *defaultResourceBackupper) backupResource(group *metav1.APIResourceList
-			}
+func (rb *defaultResourceBackupper) backupItems(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	itemBackupper ItemBackupper,
+	items []runtime.Object,
+) {
+	for _, item := range items {
+		unstructured, ok := item.(runtime.Unstructured)
+		if !ok {
+			log.Errorf("Unexpected type %T", item)
+			continue
+		}
 
-			err = itemBackupper.backupItem(log, unstructured, gr)
-			if aggregate, ok := err.(kubeerrs.Aggregate); ok {
-				log.WithField("name", metadata.GetName()).Infof("%d errors encountered backup up item", len(aggregate.Errors()))
-				// log each error separately so we get error location info in the log, and an
-				// accurate count of errors
-				for _, err = range aggregate.Errors() {
-					log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
-				}
+		metadata, err := meta.Accessor(unstructured)
+		if err != nil {
+			log.WithError(errors.WithStack(err)).Error("Error getting a metadata accessor")
+			continue
+		}
 
-				continue
-			}
-			if err != nil {
+		if gr == kuberesource.Namespaces && !rb.backupRequest.NamespaceIncludesExcludes.ShouldInclude(metadata.GetName()) {
+			log.WithField("name", metadata.GetName()).Info("Skipping namespace because it's excluded")
+			continue
+		}
+
+		err = itemBackupper.backupItem(log, unstructured, gr)
+		if aggregate, ok := err.(kubeerrs.Aggregate); ok {
+			log.WithField("name", metadata.GetName()).Infof("%d errors encountered backup up item", len(aggregate.Errors()))
+			// log each error separately so we get error location info in the log, and an
+			// accurate count of errors
+			for _, err = range aggregate.Errors() {
 				log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
-				continue
 			}
+
+			continue
+		}
+		if err != nil {
+			log.WithError(err).WithField("name", metadata.GetName()).Error("Error backing up item")
+			continue
 		}
 	}
+}
 
-	return nil
+// Adds CRD to the backup if one is found corresponding to this resource
+func (rb *defaultResourceBackupper) backupCRD(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	itemBackupper ItemBackupper,
+) {
+	crdGr := schema.GroupResource{Group: apiextv1beta1.GroupName, Resource: "customresourcedefinitions"}
+	crdClient, err := rb.dynamicFactory.ClientForGroupVersionResource(apiextv1beta1.SchemeGroupVersion,
+		metav1.APIResource{
+			Name:       "customresourcedefinitions",
+			Namespaced: false,
+		},
+		"",
+	)
+	if err != nil {
+		log.WithError(err).Error("Error getting dynamic client for CRDs")
+		return
+	}
+
+	unstructuredCRDList, err := crdClient.List(metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", gr.String())})
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error listing CRDs")
+		return
+	}
+	crds, err := meta.ExtractList(unstructuredCRDList)
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error extracting CRD list")
+		return
+	}
+	rb.backupItems(log, crdGr, itemBackupper, crds)
```

Comment:
Let's add an info log here to say we found an associated CRD to backup for the gr